ibis/sql/alchemy.py

```diff
@@ -879,26 +880,26 @@ def __init__(self, *args, **kwargs):
         self.exists = kwargs.pop('exists', False)
         super(AlchemySelect, self).__init__(*args, **kwargs)
 
-    def compile(self):
+    def compile(self, params=None):
```

Comment:
`self.params` instead? It might be simpler to have

`bound_ddl = ddl.with_params(new_params)` then having the parameter threaded everywhere